### PR TITLE
Update to embedded-hal 1.0.0

### PR DESCRIPTION
**Pull request by nik012003**
Feats:
- update embedded-hal to 1.0.0
- update display-interface to 0.5.0
- update the examples accordingly
- added embedded-hal-bus to handle spi connection

Note on examples:
- Removed with_shared_bus example since version 0.5.0 of display-interface-spi
encourages the use of embedded-hal-bus, which takes care of the bus sharing.
- I don't really like that creating a SpiDevice requires passing a Timer, further
inspection is required.

I didn't manage to test the examples with an nrf board, but I tested the changes on my esp-32c3 board using `esp-hal`.
Since the update requires the users to update to a new embedded_hal version, I think the version should be updated accordingly (i.e. bumping this crate to 0.6.0).

### CODE REVIEW

**Comment by Finomnis** [comment]
You can go ahead and bump the version :)

**Comment by Finomnis** [comment]
Please elaborate your concerns about the timer.

**Comment by Finomnis** [comment]
I'm not sure we need waiting anywhere in the transfer, maybe [`ExclusiveDevice::new_no_delay`](https://docs.rs/embedded-hal-bus/latest/embedded_hal_bus/spi/struct.ExclusiveDevice.html#method.new_no_delay) would work.

In an actual proper project this would not matter, almost all monotonics in RTIC are shareable between peripherals. So this would not utilize a second timer.

File: examples/nrf52840_dogm132w5_graphics.rs [hunk 1]
```diff
@@ -59,14 +62,19 @@ fn main() -> ! {
...
-        disp_a0,
-        disp_cs,
     );
 
+    // Create an ExclusiveDevice, the bus will be owned only by the device
+    let disp_device = ExclusiveDevice::new(spi_bus, disp_cs, hal::timer::Timer::new(peripherals.TIMER1)).unwrap();
+
+    // if you need to share the spi bus, create a shared bus and a shared device (AtomicDevice)
+    // let atomic_spi_bus = AtomicCell::new(spi_bus);
+    // let disp_device = AtomicDevice::new(&atomic_spi, disp_cs, hal::timer::Timer::new(peripherals.TIMER1)).unwrap();
```
Comment (Finomnis) [inline examples/nrf52840_dogm132w5_graphics.rs]: atomic_spi_bus

**Comment by nik012003** [comment]
Thank you! I've applied the changes, it should be ready to merge now

File: examples/nrf52840_dogm132w5_graphics.rs [hunk 2]
```diff
@@ -48,25 +52,29 @@ fn main() -> ! {
...
+        hal::spim::MODE_3,
+        0,
     );
 
+    // Create an ExclusiveDevice, the bus will be owned only by the device
+    let disp_device = ExclusiveDevice::new_no_delay(spi_bus, disp_cs).unwrap();
+
+    // if you need to share the spi bus, create a shared bus and a shared device (AtomicDevice)
+    // let atomic_spi_bus = AtomicCell::new(spi_bus);
+    // let disp_device = AtomicDevice::new(&atomic_spi_bus, disp_cs, hal::timer::Timer::new(peripherals.TIMER1)).unwrap();
```
Comment (Finomnis) [inline examples/nrf52840_dogm132w5_graphics.rs]: Use the no_delay version here as well

Comment (Finomnis) [reply examples/nrf52840_dogm132w5_graphics.rs]: Actually, I would just remove the entire AtomicDevice code. Documenting bus sharing is responsibility of `embedded_hal_bus`, not of this crate.

**Comment by Finomnis** [comment]
Min rust version got changed with `display-interface 0.5.0`

**Comment by nik012003** [comment]
Oops, I didn't catch that. Updated it

**Comment by Finomnis** [comment]
You did not test the examples yet, right?

**Comment by nik012003** [comment]
> You did not test the examples yet, right?

 Unfortunately I have an nrf board here, but I forgot my jlink at home so I can only test it in a few weeks. They do compile fine tho, and practically the same code (apart from the spi_bus initialization) works with my esp32c3.

**Comment by Finomnis** [comment]
Works on my board.

**Comment by Finomnis** [comment]
Missing: Doc updates (Readme.md)

**Comment by Finomnis** [comment]
I've fixed the README code and did some minor beautifications in the examples, are you alright with my changes?

**Pull request merged**